common/containers/Tabs/Contracts/components/withTx.tsx

```diff
@@ -4,12 +4,12 @@ import { GWei, Wei, Ether } from 'libs/units';
 import { connect } from 'react-redux';
 import { showNotification, TShowNotification } from 'actions/notifications';
 import { broadcastTx, TBroadcastTx } from 'actions/wallet';
-import { IWallet } from 'libs/wallet/IWallet';
+import { IFullWallet } from 'libs/wallet/IWallet';
 import { RPCNode } from 'libs/nodes';
 import { NodeConfig, NetworkConfig } from 'config/data';
 
 export interface IWithTx {
-  wallet: IWallet;
+  wallet: IFullWallet;
```

Comment:
@HenryNguyen5 Despite saying that this should only receive an `IFullWallet`, it's still able to receive an `IReadOnlyWallet` (Since that's what the redux store injects, any `IWallet`.) I'm quite confused about how this component works, and why it's not getting caught by the typechecker. Got any thoughts?

Reply:
Hm, looks like we've been typing these up wrong.
It should be 
```ts
  wallet: AppState['wallet']['inst'];
```

Reply:
This will throw the correct error on the files since you're grabbing AppState interface and specifying the exact type it'll be

Reply:
I think what I would prefer here is to keep `IFullWallet` as the type, and just enforce that this component only renders with an `IFullWallet`, rather than having every wallet component have some conditional in it like
```
if (!this.props.wallet || this.props.wallet.isReadOnly) {
  return null;
}
```
along with every callback function that uses full wallet only functions. Maybe this could be done with a higher order component?

Reply:
Yep, HOC or Render callback would be perfect here

Reply:
Implemented this in #387 over here: https://github.com/MyEtherWallet/MyEtherWallet/pull/387/files#diff-931b59e1826e8a1147efbbd7c1384306